test/app-luatest/http_client_test.lua

```diff
@@ -1352,3 +1353,47 @@ g.test_http_client_io_post_parts = function(cg)
     t.assert_equals(io.status, 200, 'io')
     t.assert_equals(io.reason, 'Ok', '200 - Ok')
 end
+
+-- Read from ph:stderr and fail if it doesn't contain a `pattern'.
+local function grep_stderr_or_fail(ph, pattern)
+    local output = ''
+    t.helpers.retrying({}, function()
+        local chunk = ph:read({timeout = 0.05, stderr = true})
+        if chunk ~= nil then
+           output = output .. chunk
+        end
+        t.assert_not_str_contains(output, pattern)
+    end)
+end
+
+g.test_http_client_gc_finalizer_gh_9453 = function(cg)
+    local tarantool_exe = arg[-1]
+    local lua_code = [[local url = '%s'
+                       local c = require('http.client').new()
+                       local r = c:get(url, {chunked = true})
+
+                       c = nil
+                       collectgarbage()
+                       collectgarbage()
+
+                       r:read(1)]]
```

Comment:
It's better to check result of this read because error message can be renamed. You test will check nothing.

Reply:
Updated:

```diff
diff --git a/test/app-luatest/http_client_test.lua b/test/app-luatest/http_client_test.lua
index 66f8d52c12..e684296d25 100644
--- a/test/app-luatest/http_client_test.lua
+++ b/test/app-luatest/http_client_test.lua
@@ -1354,19 +1354,11 @@ g.test_http_client_io_post_parts = function(cg)
     t.assert_equals(io.reason, 'Ok', '200 - Ok')
 end
 
--- Read from ph:stderr and fail if it doesn't contain a `pattern'.
-local function grep_stderr_or_fail(ph, pattern)
-    local output = ''
-    t.helpers.retrying({}, function()
-        local chunk = ph:read({timeout = 0.05, stderr = true})
-        if chunk ~= nil then
-           output = output .. chunk
-        end
-        t.assert_not_str_contains(output, pattern)
-    end)
-end
-
 g.test_http_client_gc_finalizer_gh_9453 = function(cg)
+    -- Nothing specific for a socket family here,
+    -- but sock_family == AF_UNIX will require a bit more additional code,
+    -- so skipped.
+    t.skip_if(cg.params.sock_family == 'AF_UNIX')
     local tarantool_exe = arg[-1]
     local lua_code = [[local url = '%s'
                        local c = require('http.client').new()
@@ -1376,13 +1368,17 @@ g.test_http_client_gc_finalizer_gh_9453 = function(cg)
                        collectgarbage()
                        collectgarbage()
 
-                       r:read(1)]]
+                       local ok, res = pcall(r.read, r, 1)
+                       assert(ok == true)
+                       assert(res ~= nil)
+                       ]]
     local ph = popen.new({ tarantool_exe, '-e', lua_code:format(cg.url) },
                          { stderr = popen.opts.PIPE })
     t.assert(ph)
 
-    -- Wait for the process startup.
-    grep_stderr_or_fail(ph, "IllegalParams: io: request must be io")
+    local stderr = ph:read({timeout = 0.05, stderr = true})
+    t.assert_equals(stderr, nil)
+    t.assert_covers(ph:wait(), {exit_code = 0})
 
     ph:close()
 end
```